### DISCUSSION TITLE
Can / should importing a constant string from an untyped library infer the type?

### DISCUSSION THREAD

**Discussion by davetapley** [microsoft/pylance-release#5678] [Ideas]
I'm using the [Falcon](https://github.com/falconry/falcon/) package which does **not** have types (yet https://github.com/falconry/falcon/issues/1820).

Accordingly I'm adding my own `.pyi` as I go, but I just came upon a weird case.

Falcon has a [constants](https://github.com/falconry/falcon/blob/a78cfb38a0c0f6031cc3ff39ff8bf4afd03ef008/falcon/constants.py#L115-L121) module which has e.g.:
```py
MEDIA_JSON = 'application/json'
MEDIA_TEXT = 'text/plain; charset=utf-8'
```

and I can import like this:

```py
from falcon.constants import MEDIA_JSON, MEDIA_TEXT
```

But VSCode is still reporting `(import) MEDIA_JSON: Unknown`.

Can it really not even infer the type of that without a hint?
I can add to my `.pyi`, but it seems crazy that it couldn't even infer that? 🤔

**Comment by erictraut** [comment]
Can you describe your configuration in more detail? Is the `from falcon.constants...` statement located in a ".py" file within your project, or are you trying to import (and then re-export) these symbols from one of your ".pyi" files?

If I install the `falcon` library and then create a "dummy.py" file and paste in your import statement, here's what I see for the type inference:

<img width="440" alt="image" src="https://github.com/microsoft/pylance-release/assets/7040122/f53eb366-06e0-40e7-9647-e317598e3bfe">
